packages/amplify-provider-awscloudformation/lib/build-resources.js

```diff
@@ -14,134 +11,90 @@ async function run(context, category, resourceName) {
-  const scriptName = `amplify:${resourceName}`;
-  if (scriptExists(projectRoot, scriptName)) {
-    runPackageManager(projectRoot, scriptName);
+  // package the function
+  const distDir = path.join(resourcePath, 'dist');
+  if (!fs.existsSync(distDir)) {
+    fs.mkdirSync(distDir);
   }
-}
-
-function scriptExists(projectRoot, scriptName) {
-  const packageJsonPath = path.normalize(path.join(projectRoot, 'package.json'));
-  if (fs.existsSync(packageJsonPath)) {
-    const rootPackageJsonContents = require(packageJsonPath);
-    return rootPackageJsonContents.scripts && rootPackageJsonContents.scripts[scriptName];
+  const destination = path.join(distDir, 'latest-build.zip');
+  let packagePromise;
+  if (breadcrumbs.scripts && breadcrumbs.scripts.package) {
+    // TODO
+    throw new Error('Executing custom package scripts is not yet implemented');
+  } else {
+    const packageRequest = {
+      env: context.amplify.getEnvInfo().envName,
+      srcRoot: resourcePath,
+      dstFilename: destination,
+      runtime: breadcrumbs.functionRuntime,
+      lastPackageTimestamp: resource.lastPackageTimestamp ? new Date(resource.lastPackageTimestamp) : undefined,
+      lastBuildTimestamp: rebuilt ? new Date() : new Date(resource.lastBuildTimeStamp),
+    };
+    packagePromise = runtimePlugin.package(packageRequest);
   }
-  return false;
-}
-
-function installDependencies(resourceDir) {
-  runPackageManager(resourceDir);
-}
-
-function runPackageManager(cwd, scriptName = undefined) {
-  const isWindows = /^win/.test(process.platform);
-  const npm = isWindows ? 'npm.cmd' : 'npm';
-  const yarn = isWindows ? 'yarn.cmd' : 'yarn';
-  const useYarn = fs.existsSync(`${cwd}/yarn.lock`);
-  const packageManager = useYarn ? yarn : npm;
-  const args = toPackageManagerArgs(useYarn, scriptName);
-  const childProcessResult = childProcess.spawnSync(packageManager, args, {
-    cwd,
-    stdio: 'pipe',
-    encoding: 'utf-8',
+  return new Promise((resolve, reject) => {
+    packagePromise
+      .then(result => {
+        const packageHash = result.packageHash;
+        zipFilename = packageHash ? `${resource.resourceName}-${packageHash}-build.zip` : zipFilename;
+        context.amplify.updateAmplifyMetaAfterPackage(resource, zipFilename);
+        resolve({ zipFilename, zipFilePath: destination });
+      })
+      .catch(err => reject(new Error(`Package command failed with error [${err}]`)));
   });
-  if (childProcessResult.status !== 0) {
-    throw new Error(childProcessResult.output);
-  }
 }
 
-function toPackageManagerArgs(useYarn, scriptName) {
-  if (scriptName) {
-    return useYarn ? [scriptName] : ['run-script', scriptName];
+async function loadRuntimePlugin(context, pluginId) {
+  const pluginMeta = context.pluginPlatform.plugins.functionRuntime.find(meta => meta.manifest.functionRuntime.pluginId === pluginId);
+  if (!pluginMeta) {
+    throw new Error(`Could not find runtime plugin with id [${pluginId}] to build the resource`)
```

Comment:
to build the resource: {resourceName} add the name of the function to provide a better error message to our customers.